### DISCUSSION TITLE
Is it possible to only update one flake input with Nix?

### DISCUSSION THREAD

**Discussion by EricCrosson** [renovatebot/renovate#19582] [Request Help]
## Overview

I have a NixOS flake with [numerous inputs](https://github.com/EricCrosson/dotfiles/blob/78a6931fb3ad779f80bde75655cc2849a108f72d/flake.nix#L56). I'm trying to configure Renovate to **ignore all inputs except for one**, called `helix`. I understand this is possible with direct dependencies using other managers, but I have limited experience with `lockFileMaintenance`.

Am I trying to attempt something that's currently supported? Are there plans to support this use case?

## What I did

Using [this renovate.json config](https://github.com/EricCrosson/dotfiles/blob/78a6931fb3ad779f80bde75655cc2849a108f72d/.github/renovate.json), I extended the minimal nix configuration with `packageRules` to disable all `nix` updates except for the package pattern` helix`.

## What I expected to happen

I was expecting Renovate to open a PR updating only the `helix` input in the `flake.lock` file, like [this](https://github.com/EricCrosson/dotfiles/pull/259) (which is not created by Renovate).

## What actually happened

Renovate opened a PR updating _every_ input in the `flake.lock` file [here](https://github.com/EricCrosson/dotfiles/pull/261).

I'm not sure if this is relevant, but it looks like Renovate is not detecting specific nix dependencies (see picture). I suppose this may be expected since I have to use `lockFileMaintenance` to update flake inputs.

![image](https://user-images.githubusercontent.com/1596818/209580399-150ca08b-7953-41c2-9c41-e5cfec042cdb.png)

(This is an empty list.)

## Non sequitur

I'm ecstatic to see Nix support in Renovate! Thank you so much for your work on this great tool and congratulations on the recent beta launch of Nix support :rocket:

**Comment by eliasp** [comment]
Nix itself added support for updating a specific input of a Flake in https://github.com/NixOS/nix/commit/b9fb3720750a24d8b2fbe6b1491696d6a51dcff3, using e.g. `nix flake lock --update-input helix`.

I suppose Renovate uses Nix under the hood and is therefore depending on this capability.
